app/sockets.py

```diff
@@ -0,0 +1,153 @@
+from .__init__ import *
+
+
+from .MyFunctions import *
+# from .models import *
+from .schemas import *
+from .routes import *
+
+
+@socketio.on('join', namespace='/rooms_namespace')
+def on_join(data):
+
+    """
+    verify if attendence is success 
+    then get them join in that room name with otp added
+    now broadcast to that room
+    """
+    try:
+        data = json.loads(data) # now it's dict form, before it was string
+    except:
+        data = {}
+
+    status = data.get('status')
+    if status!='Success':
+        return ''
+    
+    room = '-'.join(data.get('event_name').split(' ')) + '-' + str(data['otp'])
+    username = data.get("username") #add this to payLoad (clientside)
+    join_room(room)
+    mymessage = {
+        'username':data.get('username'),
+        'RoomName':room
+    }
+
+    emit('join_room', json.dumps(mymessage), room=room)
+
+
+
+@socketio.on('admin_stats', namespace='/admin_namespace')
+def made_for_admin():
+
+    """
+    To tell flask-SocketIO that there exsists a namespace named admin_namespace
+    then we can use it in functions written below. Or else error will be thrown.
+    """
+    #admin_stats is mapped with admin_listen
+    pass
+
+
+
+@socketio.on('attendence_request', namespace='/attendence_namespace')
+def take_attendence_from_user(json_):
+    
+    #attendence_request is mapped with attendence_result at now
+    """
+    This will mark attendence if valid or return failure reason
+    at the namespace 'attendence_namespace' and this is done privately for
+    the user.
+    It also broadcasts the attendence result to
+    """
+
+    otp_ = json_['otp']
+    token_ = json_['token']
+
+
+    user_detail = user_info(token_)
+    if user_detail != 'AuthFail':
+        
+        email_ = user_info(token_).get("email")
+        datetime_ = datetime.datetime.now()
+        # workleft: In client payLoad take latitude and longitude of user and then
+        # compare it with admins lat and long and judge on distance param
+        # Make change after making change in events model
+        status_ = 1 # code logic needed here
+        try:
+            event_id_ = Events.query.filter_by(otp=otp_).first().id
+
+        except Exception as e:
+            event_id_ = -1
+
+        if event_id_>=0:
+
+            new_attendence = Attendence(event_id=event_id_, event_otp=otp_, email=email_, \
+                datetime=datetime_, status=status_)
+            db.session.add(new_attendence)
+            
+            if status_==0:
+                payLoad = {
+                    'Status':'Fail',
```

Comment:
Use status code rather than messages

Reply:
I have used status code for all apis, but left for sockets.
I will check for if sockets support make_response as they are emiting not returning for our case.
Otherwise I will internally implement this, kept it in future improvements.

Reply:
https://github.com/GDGVIT/AttendanceApp-Backend/issues/10#issue-559113225